BibBuddy/app/src/main/java/de/bibbuddy/BookFragment.java

```diff
@@ -203,80 +240,169 @@ public void onClick(DialogInterface dialog, int which) {
     }
   }
 
-  private void checkStoragePermission() {
-    if (ContextCompat.checkSelfPermission(getContext(),
-        Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
-      requestStoragePermission();
+  ActivityResultLauncher<Intent> filePickerActivityResultLauncher = registerForActivityResult(
+      new ActivityResultContracts.StartActivityForResult(),
+      new ActivityResultCallback<ActivityResult>() {
+        @Override
+        public void onActivityResult(ActivityResult result) {
+          if (result.getResultCode() == Activity.RESULT_OK) {
+            Intent data = result.getData();
+
+            if (data != null) {
+
+              Uri uri = data.getData();
+              if (importBibTex.isBibFile(UriUtils.getFullUriPath(context, uri))) {
+
+                handleImport(uri);
+
+                Toast.makeText(context, getString(R.string.imported_file_name_is) + '\n'
+                    + UriUtils.getUriFileName(getActivity(), uri), Toast.LENGTH_LONG).show();
+
+              } else {
+                showDialogNonBibFile();
```

Comment:
```suggestion
            if (data == null) {
              showDialogNonBibFile();
         
              } else {
                  Uri uri = data.getData();
              if (importBibTex.isBibFile(UriUtils.getFullUriPath(context, uri))) {

                handleImport(uri);

                Toast.makeText(context, getString(R.string.imported_file_name_is) + '\n'
                    + UriUtils.getUriFileName(getActivity(), uri), Toast.LENGTH_LONG).show();

```

Reply:
This is the same as my approach. I will not change it, because according to the documentation of Android the first case is also data!=null.